server/routes/pollRouter.js

```diff
@@ -86,4 +90,72 @@ router.delete('/delete', auth, async (req, res) => {
     }
 });
 
+router.put('/vote-one-add', auth, async (req, res) => {
+    try {
+        const { pollId, imageOneVotes } = req.body;
+
+        const updatedPoll = await Poll.findById(pollId);
+        updatedPoll.imageOneVotes++;
+        updatedPoll.votedForImageOne.push(req.user);
+        await updatedPoll.save();
+
+        res.json(updatedPoll);
+    } catch (err) {
+        res.status(500).json({ error: err.message });
+    }
+});
+
+router.put('/vote-one-remove', auth, async (req, res) => {
+    try {
+        const { pollId, imageOneVotes } = req.body;
+
+        const updatedPoll = await Poll.findById(pollId);
+        updatedPoll.imageOneVotes--;
+        await updatedPoll.save();
+
+        await Poll.updateOne(
+            { '_id': pollId }, 
+            { $pull: { 'votedForImageOne' : req.user } }
+        );
+
+        res.json(updatedPoll);
+    } catch (err) {
+        res.status(500).json({ error: err.message });
+    }
+});
+
+router.put('/vote-two-add', auth, async (req, res) => {
```

Comment:
same thing here, adding and removing controllers generally do the same thing. the logic may be different by can be decided depending on its parameters

Reply:
this would reduce repeated code. in addition, if in the future we were to introduce a 3rd option, it would be easily scalable

Reply:
refactored. not sure why I didn't do it right away :)